grakn-graql/src/main/java/ai/grakn/graql/internal/pattern/property/RelatesProperty.java

```diff
@@ -96,12 +115,23 @@ public String getProperty() {
         PropertyExecutor relatesExecutor = PropertyExecutor.builder(relatesMethod).requires(var, roleVar).build();
 
         // This allows users to skip stating `$roleVar sub role` when they say `$var relates $roleVar`
-        PropertyExecutor.Method isRoleMethod = executor -> {
-            executor.builder(roleVar).isRole();
-        };
+        PropertyExecutor.Method isRoleMethod = executor -> executor.builder(roleVar).isRole();
 
         PropertyExecutor isRoleExecutor = PropertyExecutor.builder(isRoleMethod).produces(roleVar).build();
 
+        VarPatternAdmin superRoleVarPattern = superRole();
+        if (superRoleVarPattern != null) {
+            Var superRoleVar = superRoleVarPattern.var();
+            PropertyExecutor.Method subMethod = executor -> {
+                Role superRole = executor.get(superRoleVar).asRole();
+                executor.builder(roleVar).sub(superRole);
+            };
+
+            PropertyExecutor subExecutor = PropertyExecutor.builder(subMethod)
+                    .requires(superRoleVar).produces(roleVar).build();
+
+            return ImmutableSet.of(relatesExecutor, isRoleExecutor, subExecutor);
+        }
         return ImmutableSet.of(relatesExecutor, isRoleExecutor);
```

Comment:
It might be a bit neater to make this an `if ... else`:
```
if (superRoleVarPattern != null) {
    ...
    return ImmutableSet.of(relatesExecutor, isRoleExecutor, subExecutor);
} else {
    return ImmutableSet.of(relatesExecutor, isRoleExecutor);
}
```

but maybe it's just personal preference 😛